### ISSUE TITLE
tmt clean guests -> TypeError: expected string or bytes-like object

### ISSUE THREAD

**Issue by lukaszachy** [teemtee/tmt#1311]
`tmt clean guests` fails with traceback, debugging shows:
```
$ python3 -m pdb `which tmt` clean guests
> /usr/bin/tmt(3)<module>()
-> import click
(Pdb) c
clean
    guests
Traceback (most recent call last):
  File "/usr/lib64/python3.10/pdb.py", line 1726, in main
    pdb._runscript(mainpyfile)
  File "/usr/lib64/python3.10/pdb.py", line 1586, in _runscript
    self.run(statement)
  File "/usr/lib64/python3.10/bdb.py", line 597, in run
    exec(cmd, globals, locals)
  File "<string>", line 1, in <module>
  File "/usr/bin/tmt", line 8, in <module>
    tmt.cli.main()
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1690, in invoke
    rv.append(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3.10/site-packages/tmt/cli.py", line 1196, in guests
    if not tmt.Clean(parent=context.obj.clean, context=context).guests():
  File "/usr/lib/python3.10/site-packages/tmt/base.py", line 1984, in guests
    if not self._stop_running_guests(run):
  File "/usr/lib/python3.10/site-packages/tmt/base.py", line 1949, in _stop_running_guests
    plan.provision.wake()
  File "/usr/lib/python3.10/site-packages/tmt/steps/provision/__init__.py", line 85, in wake
    plugin.wake(data=self._guest_data.get(plugin.name))
  File "/usr/lib/python3.10/site-packages/tmt/steps/provision/testcloud.py", line 242, in wake
    guest.wake()
  File "/usr/lib/python3.10/site-packages/tmt/steps/provision/testcloud.py", line 415, in wake
    self._image = testcloud.image.Image(self.image_url)
  File "/usr/lib/python3.10/site-packages/testcloud/image.py", line 72, in __init__
    uri_data = self._process_uri(uri)
  File "/usr/lib/python3.10/site-packages/testcloud/image.py", line 93, in _process_uri
    type_match = re.search(r'(http|https|file)://([\w\.\-/]+)', uri)
  File "/usr/lib64/python3.10/re.py", line 200, in search
    return _compile(pattern, flags).search(string)
TypeError: expected string or bytes-like object
Uncaught exception. Entering post mortem debugging
Running 'cont' or 'step' will restart the program
> /usr/lib64/python3.10/re.py(200)search()
-> return _compile(pattern, flags).search(string)
(Pdb) p string
None
(Pdb) up
> /usr/lib/python3.10/site-packages/testcloud/image.py(93)_process_uri()
-> type_match = re.search(r'(http|https|file)://([\w\.\-/]+)', uri)
(Pdb) up
> /usr/lib/python3.10/site-packages/testcloud/image.py(72)__init__()
-> uri_data = self._process_uri(uri)
(Pdb) up
> /usr/lib/python3.10/site-packages/tmt/steps/provision/testcloud.py(415)wake()
-> self._image = testcloud.image.Image(self.image_url)
(Pdb) p self.image_url
None
(Pdb) p self.workdir
'/var/tmp/tmt/run-064/plans/default/provision/default'
(Pdb) 
```

File '/var/tmp/tmt/run-064/plans/default/provision/default/guests.yaml' contains
```
default:
    guest: 127.0.0.1
    port: 10050
    user: root
    key: /var/tmp/tmt/run-064/plans/default/provision/default/id_ecdsa
    instance: tmt-064-QRYUVmtO
    image: file:///var/tmp/tmt/testcloud/images/rhel-guest-image-8.4-1255.x86_64.qcow2
    connection: session
    arch: x86_64
```

**Comment by idorax** [comment]
There are two lines in [tmt/steps/provision/testcloud.py](https://github.com/teemtee/tmt/blob/main/tmt/steps/provision/testcloud.py), 
```python
    def wake(self):
        """ Wake up the guest """
        ...<snip>...
        self._image = testcloud.image.Image(self.image_url)

    def start(self):
        ...<snip>...
        self._image = testcloud.image.Image(self.image_url)
```

Since `testcloud.image.Image()` is out of TMT's control, looks we have to make sure `image_url` is not `None` before invoking  `testcloud.image.Image()` , right? @lukaszachy

**Comment by psss** [comment]
I believe the original issue has been fixed by #1247 which ensures all keys are correctly saved to `guests.yaml` and restored during the `wake()` up. Feel free to reopen if the issue reappears.
